Tests/UnitTests/AdyenTheme/AdyenThemeTextFieldTests.swift

```diff
@@ -0,0 +1,198 @@
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.cornerRadius, cornerRadius)
+    }
+    
+    func test_textFieldMethod_shouldUpdateBorderWidth() {
+        let expectedBorderWidth: CGFloat = 2.0
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.borderWidth = expectedBorderWidth
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.borderWidth, expectedBorderWidth)
+    }
+    
+    func test_textFieldMethod_shouldUpdateTitleStyle() {
+        // Given
+        let theme = AdyenTheme()
+        
+        let newTitleStyle = AdyenLabelStyle(
+            font: .preferredFont(forTextStyle: .title1),
+            color: .red,
+            textAlignment: .natural
+        )
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.title = newTitleStyle
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.title.color, .red)
+        XCTAssertEqual(updatedTheme.textFieldStyle.title.font, .preferredFont(forTextStyle: .title1))
+        XCTAssertEqual(updatedTheme.textFieldStyle.title.textAlignment, .natural)
+    }
+    
+    func test_textFieldMethod_shouldUpdateTextStyle() {
+        // Given
+        let theme = AdyenTheme()
+        
+        let newTextStyle = AdyenLabelStyle(
+            font: .preferredFont(forTextStyle: .title2),
+            color: .systemPink,
+            textAlignment: .left
+        )
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.text = newTextStyle
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.text.color, .systemPink)
+        XCTAssertEqual(updatedTheme.textFieldStyle.text.font, .preferredFont(forTextStyle: .title2))
+        XCTAssertEqual(updatedTheme.textFieldStyle.text.textAlignment, .left)
+    }
+    
+    func test_textFieldMethod_shouldUpdatePlaceholderStyle() {
+        // Given
+        let theme = AdyenTheme()
+        
+        let newPlaceholderTextStyle = AdyenLabelStyle(
+            font: .preferredFont(forTextStyle: .footnote),
+            color: .red,
+            textAlignment: .right
+        )
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.placeholderText = newPlaceholderTextStyle
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.placeholderText?.color, .red)
```

Comment:
It is better to validate against the values you passed. This will make this code easier to maintain and change in the future.
`XCTAssertEqual(updatedTheme.textFieldStyle.placeholderText?.color, newPlaceholderTextStyle.color)`